sentinelhub/data_utils.py

```diff
@@ -24,73 +25,62 @@ def _extract_hist(hist_cols: Tuple[str, str], hist_data: List[JsonDict]) -> Dict
             bins.append(hist_data[idx]["highEdge"])
         counts.append(hist_data[idx]["count"])
 
-    return {hist_cols[0]: bins, hist_cols[1]: counts}
-
-
-def _extract_percentiles(col_name: str, perc_data: Dict[str, float]) -> Dict[str, float]:
-    """Transform statistical percentiles into pandas.DataFrame entry
-
-    :param col_name: Output name and band name. Example: ndvi_B0.
-    :param perc_data: An input representation of percentile data of <col_name>.
-    :return: Percentile data as a pandas.DataFrame entry. Examples: {"ndvi_B0_<perc>": perc_val}.
-    """
-    perc_entry = {}
-    for perc, perc_val in perc_data.items():
-        perc_col_name = f"{col_name}_{perc}"
-        perc_entry[perc_col_name] = perc_val
-    return perc_entry
+    return bins, counts
 
 
-def _extract_stats(
-    interval_output: JsonDict, excl_stats: List[str], incl_histogram: bool
-) -> Dict[str, Union[List[float], float]]:
+def _extract_stats(interval_output: JsonDict, exclude_stats: List[str]) -> Dict[str, Union[List[float], float]]:
     """Transform statistics into pandas.DataFrame entry
 
     :param interval_output: An input representation of statistics of an aggregation interval.
-    :param excl_stats: Unwanted statistical name.
-    :param incl_histogram: Flag to transform histogram.
+    :param exclude_stats: Statistics that will be excluded from output.
     :return: Statistics as a pandas.DataFrame entry.
     """
     stat_entry: Dict[str, Union[List[float], float]] = {}
     for output_name, output_data in interval_output.items():
         for band_name, band_values in output_data["bands"].items():
             band_stats = band_values["stats"]
+            # statistics are not valid when sample count equals to no data count
             if band_stats["sampleCount"] == band_stats["noDataCount"]:
                 break
 
             for stat_name, value in band_stats.items():
-                if stat_name not in excl_stats:
+                if stat_name not in exclude_stats:
                     col_name = f"{output_name}_{band_name}_{stat_name}"
                     if stat_name == "percentiles":
-                        stat_entry.update(_extract_percentiles(col_name, value))
+                        stat_entry.update(
+                            {
+                                f"{col_name}_{percentile_name}": percentile_value
+                                for percentile_name, percentile_value in value.items()
+                            }
+                        )
```

Comment:
Would this be shorter if you did it manually (without update), e.g.
```python
for percentile_name, percentile_value in value.items():
    stat_entry[f"..."] = percentile_value
```
?

The same should be tried later for histogram values

Reply:
If a for loop is added here there's a complain of `too many nested clocks (6/5)` from mypy. I also think the original styling in the [helper function](https://sentinelhub-py.readthedocs.io/en/latest/examples/statistical_request.html#Helper-function) is better. 

Any suggested workaround in this case?

Reply:
I would prefer disabling the linter with an inline comment (it's probably pylint, so something along `# pylint: disable=<error-code>`)
because this way you have more indentation and a for loop remains.